aldryn_events/models.py

```diff
@@ -194,13 +194,26 @@ def clean(self):
                 )
             )
 
-        if (self.start_date and self.end_date and
-                self.end_date < self.start_date):
-            raise ValidationError(_('Start date should be before end date.'))
-
-        if (self.end_date and self.start_date == self.end_date and
-                self.end_time < self.start_time):
-            raise ValidationError(_('Start date should be before end date.'))
+        # there is a start date and end date
+        if self.start_date and self.end_date:
+            if self.end_date < self.start_date:
+                raise ValidationError(_('Start date should be before end date.'))
+
+            # dates are equal, check time
+            if self.start_date == self.end_date:
+                # check that time is provided
+                if not (self.end_time and self.start_time):
+                    raise ValidationError(
+                        _('If you specify same dates for start and end of event'
+                          ' please provide also start and end time. OR you can'
+                          ' make end date to be different'))
+
+                # check time validity
+                if (self.end_time < self.start_time or
+                            self.start_time == self.end_time):
```

Comment:
weird indentation. the second line of the conditional should align with the first.